### PR TITLE
fix(linter): ensure flat config generator works for pcv3 plugin

### PR DESCRIPTION
**Pull request by meeroslav**
## Related Issue(s)

Fixes #

### CODE REVIEW

File: packages/eslint/src/generators/convert-to-flat-config/generator.spec.ts [hunk 1]
```diff
@@ -430,4 +431,63 @@ describe('convert-to-flat-config generator', () => {
...
+    updateJson(
+      tree,
+      'libs/test-lib/project.json',
+      (json: ProjectConfiguration) => {
+        delete json.targets.lint;
+        return json;
+      }
+    );
+
+    await convertToFlatConfigGenerator(tree, options);
```
Comment (JamesHenry) [inline packages/eslint/src/generators/convert-to-flat-config/generator.spec.ts:456]: Please could we add an assertion just before that the flat config doesn't already exist? Verifying that mentally requires taking in the whole rest of the file right now

File: packages/eslint/src/generators/convert-to-flat-config/generator.spec.ts [hunk 2]
```diff
@@ -430,4 +431,63 @@ describe('convert-to-flat-config generator', () => {
...
+    });
+    updateJson(
+      tree,
+      'libs/test-lib/project.json',
+      (json: ProjectConfiguration) => {
+        delete json.targets.lint;
+        return json;
+      }
+    );
+
```
Comment (JamesHenry) [inline packages/eslint/src/generators/convert-to-flat-config/generator.spec.ts:488]: Same comment

File: packages/eslint/src/generators/convert-to-flat-config/generator.ts [hunk 3]
```diff
@@ -102,13 +104,20 @@ function convertProjectToFlatConfig(
         }
         updateProjectConfiguration(tree, project, projectConfig);
       }
-      const nxHasLintTargets = Object.keys(nxJson.targetDefaults || {}).some(
+      const nxHasEsLintTargets = Object.keys(nxJson.targetDefaults || {}).some(
         (t) =>
           (t === '@nx/eslint:lint' ||
-            nxJson.targetDefaults[t].executor === '@nx/eslint:lint') &&
+            nxJson.targetDefaults[t].executor === '@nx/eslint:lint' ||
+            nxJson.targetDefaults[t].command.includes('eslint ')) &&
```
Comment (JamesHenry) [inline packages/eslint/src/generators/convert-to-flat-config/generator.ts:111]: Same comment

File: packages/eslint/src/generators/convert-to-flat-config/generator.ts [hunk 4]
```diff
@@ -88,7 +88,9 @@ function convertProjectToFlatConfig(
   if (eslintFile && !eslintFile.endsWith('.js')) {
     if (projectConfig.targets) {
       const eslintTargets = Object.keys(projectConfig.targets || {}).filter(
-        (t) => projectConfig.targets[t].executor === '@nx/eslint:lint'
+        (t) =>
+          projectConfig.targets[t].executor === '@nx/eslint:lint' ||
+          projectConfig.targets[t].command.includes('eslint ')
```
[the remark this answers is not in the document]
Comment (meeroslav) [reply packages/eslint/src/generators/convert-to-flat-config/generator.ts]: I thought that having just `eslint` might have some unexpected things caught. But I guess we can solve that once it happens.

File: packages/eslint/src/generators/convert-to-flat-config/generator.ts (hunk 3 above)
Comment (meeroslav) [reply packages/eslint/src/generators/convert-to-flat-config/generator.ts]: Same as above

**Review by JamesHenry — APPROVED** [review]

**Pull request merged**